### ISSUE TITLE
[Internship Project]: Hellen Namulinda

### ISSUE THREAD

**Issue by GemmaTuron** [ersilia-os/ersilia#714]
### Summary

Hello,

This is a public issue for a virtual daily stand-up. We will use this to briefly share the tasks of the day and the challenges and advances made, so that we can ensure smooth support from the Ersilia mentors and alignment between daily tasks and overall internship goals.

### Scope

Initiative 🐋

### Objective(s)

Internship goals:

- Learn how to work in an Open Source Community
- Improve the infrastructure of the Ersilia Model Hub
- Improve the usability and platform compatibility of Ersilia Models, mainly via Docker Images
- Learn about AI/ML models for drug discovery
- Identify new relevant AI/ML models and datasets and add them to this [discussion](https://github.com/ersilia-os/ersilia/discussions/694)
- Incorporate new AI/ML models in the Hub
- Improve the documentation of the Ersilia Model Hub for users
- Improve the documentation of the Ersilia Model Hub for developers

### Team

| Role & Responsibility | Username(s) | 
|--|--|
| Intern | @HellenNamulinda |
| Mentor| @Gemma |
| Coordinator| @GemmaTuron  |

### Timeline

Before starting your work, line up a few tasks and short description. This should not take long. For example, it could be something like:
Wednesday 21st June
- [ ] Create Intern daily task tracker
- [ ] Update the Ersilia GitHub Project
- [ ] Make sure Git Actions start upon success of the other workflows
- [ ] Work on the documentation for model testing
- [ ] Attend the Drug Discovery intro by WCAIR

**Comment by HellenNamulinda** [comment]
**Wednesday 21st June:**
- Debug: [Clean UP & Dockerization eos31ve](https://github.com/ersilia-os/eos31ve/issues/4) 
- [ ] Tested model  eos81ew using Colab,  CLI and Docker: [New model ready for testing!](https://github.com/ersilia-os/eos81ew/issues/8#issuecomment-1601335110)
- [ ] Tested model  eos81ew(Partially done using Colab): [New model ready for testing!](https://github.com/ersilia-os/eos7pw8/issues/9#issuecomment-1601551005)
- [ ] Reinstalled ersilia, some models were failing on CLI when given output files. Will test model [eos9tyg](https://github.com/ersilia-os/eos9tyg/issues/4)
- [ ] Attended the Drug Discovery Intro course session
- [ ] Attended the RightsCon Takeaways for Ersilia event

Model `eos31ve` whose changes were merged, was [tested](https://github.com/ersilia-os/eos31ve/issues/6) and feedback shows it's working. Though its arm64 build failed, then amd64 was successful.
Docker images are quite big(4GB +) and my limited internet hindered the testing.

**Comment by HellenNamulinda** [comment]
**Thursday 22nd June:**
- [ ] Refactor and debug model `eos92sw`: [Clean UP & Dockerization eos92sw#3](https://github.com/ersilia-os/eos92sw/issues/3)
- [ ] Test model  `eos9tyg` using CLI and Docker: [New model ready for testing! eos9tyg#4](https://github.com/ersilia-os/eos9tyg/issues/4) 
- [ ] Test model  `eos7pw8` using CLI and Docker: [New model ready for testing! eos7pw8#9](https://github.com/ersilia-os/eos7pw8/issues/9#issuecomment-1601551005) 
- [ ] Test model  `eos526j` using Docker: [New model ready for testing! eos526j#4](https://github.com/ersilia-os/eos526j/issues/4) 
- [ ] Present progress at the Weekly Lab meeting

**Comment by HellenNamulinda** [comment]
**Friday 23rd June: Tasks**
- Model CleanUp:  
    - [ ] https://github.com/ersilia-os/eos2lqb/issues/16
    - [ ]  https://github.com/ersilia-os/eos65rt/issues/6
    - [ ] Prepare a presentation on https://github.com/ersilia-os/eos2lqb/issues/12
- Pending model testing
    -  [ ] Test model  `eos526j` using Docker: https://github.com/ersilia-os/eos526j/issues/4
    - [ ] Test model  `eos9tyg` using CLI and Docker: https://github.com/ersilia-os/eos9tyg/issues/4
    - [ ] Test model  `eos7pw8` using CLI and Docker: https://github.com/ersilia-os/eos7pw8/issues/9#issuecomment-1601551005 

I'm working to resolve the pending tasks by the end of the weekend.

**Comment by HellenNamulinda** [comment]
**Monday 26th June: Tasks**
 -  [ ] Test model  `eos2re5`: https://github.com/ersilia-os/eos2re5/issues/4  
Model CleanUp :   
    - [ ]  https://github.com/ersilia-os/eos65rt/issues/6, submitted a PR, https://github.com/ersilia-os/eos65rt/pull/9 to ressolve [run api error.](https://github.com/ersilia-os/eos65rt/issues/8)

Troubleshooting
    - [ ] Preparing a presentation on https://github.com/ersilia-os/eos2lqb/issues/12 

Testing model `eos2re5` using Colab failed, it was requiring shells to be activated. The sudo run commands also required root access when using CLI, and even after providing the password, it was taking too long to fetch the model. 

The first PR I made for refactoring model eos65rt didn't capture all the commits for updating api, and it was failing at testing using ersilia run. I pushed the changes for updating api to run.

Currently, ersilia crashes when given invalid smiles(wrong inputs). Instead of continuing to predict for the correct ones, the execution halts. The exception isn't handled correctly. 
And since most models, don't have a way of running predictions for only right smiles, this has to be handled by ersilia such that when making inferences, only valid smiles are passed. For the wrong inputs, their prediction value should a message of say "Invalid smile".  
I setup a VM to test proposed changes without affecting my normal ersilia environment.

**Comment by HellenNamulinda** [comment]
@GemmaTuron, the pending issue I'm working is the presentation and solution to 
- https://github.com/ersilia-os/eos2lqb/issues/12

**Comment by HellenNamulinda** [comment]
**Tuesday 27th June: Tasks**
- [ ] Refactor model `eos2gth`: https://github.com/ersilia-os/eos92sw/issues/3
- [ ] Re-test model  `eos7pw8` : https://github.com/ersilia-os/eos7pw8/issues/9#issuecomment-1608928079
- [ ] Test model  `eos97yu` : https://github.com/ersilia-os/eos97yu/issues/7
- [ ] Present about  https://github.com/ersilia-os/eos2lqb/issues/12

**Comment by HellenNamulinda** [comment]
**Wednesday 27th June: Tasks**
- [ ] Refactor model `eos6hy3`: [Clean UP & Dockerization eos6hy3](https://github.com/orgs/ersilia-os/projects/1/views/10?pane=issue&itemId=29234795)
- [ ] Attend coding session organized by Miquel  on https://github.com/ersilia-os/eos2lqb/issues/12

**Comment by GemmaTuron** [comment]
@HellenNamulinda 
I've added a second model in case you finish all the tasks!

**Comment by HellenNamulinda** [comment]
**Thursday 29th June: Tasks**
- [ ] Refactor model `eos3sa2`: https://github.com/ersilia-os/eos3sa2/issues/1
- [ ] Test model `eos4zfy `: https://github.com/ersilia-os/eos4zfy/issues/4 
- [ ]  Test model `eos7pw8`: https://github.com/ersilia-os/eos7pw8/issues/9#issuecomment-1611784598
- [ ] Weekly Lab Meeting

**Comment by GemmaTuron** [comment]
Hi @HellenNamulinda !

See my comment on model es3sa2, I am afraid is a work in progress model that was never finished, I am sorry for this!

**Comment by HellenNamulinda** [comment]
Hello @GemmaTuron,
Oh yes, I have read the comment.  When we are done with cleaning the other models, I will be glad to investigate the model and if possible re-incorporate it.

I will first continue to the next model(`[eos4avb](https://github.com/orgs/ersilia-os/projects/1/views/10?pane=issue&itemId=29235311)`).

Also, when I convert the suggested task to issue, it unassigns me, forexample model [eos6hy3](https://github.com/orgs/ersilia-os/projects/1/views/10?pane=issue&itemId=29234795). Hope you will be handling that.

**Comment by GemmaTuron** [comment]
Hi @HellenNamulinda 
I've fixed the task to issue thing !

**Comment by HellenNamulinda** [comment]
Hello Gemma,
Thank you!
Due to power issues today,  I'm handling the tasks late. But I will go through all my pending tasks before next week.

**Comment by HellenNamulinda** [comment]
Hi @GemmaTuron,
**Monday 3rd July: Tasks**
- [ ] Refactor model `eos4avb`: https://github.com/ersilia-os/eos4avb/issues/7
- [ ] Refactor model `eos7asg`: https://github.com/ersilia-os/eos7asg/issues/6
- [ ] Test model `eos3cf4`: https://github.com/ersilia-os/eos3cf4/issues/3

**Comment by HellenNamulinda** [comment]
Hi @GemmaTuron,
**Tuesday 4th July: Tasks**
- [ ] Debug errors to `eos7asg` testing: https://github.com/ersilia-os/eos7asg/issues/6, https://github.com/ersilia-os/eos7asg/issues/5

**Comment by GemmaTuron** [comment]
Hi @HellenNamulinda !

Good, I've assigned you a new model in case you finish with debugging the eos7asg.

**Comment by HellenNamulinda** [comment]
Sure,
I will work on it.

**Comment by GemmaTuron** [comment]
Hi @HellenNamulinda 

I have assigned you a new model in case you are dine with the eos2thm, but don't worry if you don't get to it today.

**Comment by HellenNamulinda** [comment]
**Wednesday 5th July: Tasks**
- [ ] Refactor model `eos2thm`: https://github.com/ersilia-os/eos2thm/issues/6
- [ ] Refactor model `eos8a4x`: https://github.com/ersilia-os/eos8a4x/issues/6
- [ ] Test model `eos3zur`: https://github.com/ersilia-os/eos3zur/issues/5
- [ ] Test model `eos2rd8`: https://github.com/ersilia-os/eos2rd8/issues/1

**Comment by HellenNamulinda** [comment]
Hi @GemmaTuron,
Yes I have to finish these two models for refactoring. Plus testing the one pending testing after changes by Emma.
I'm done with eos8a4x but finalizing with local testing. 
With the  [comment](https://github.com/ersilia-os/eos2thm/issues/6#issuecomment-1623085553) I just added on model eos2thm. I just need your go ahead.

**Comment by HellenNamulinda** [comment]
Hello @GemmaTuron 
### Thursday 6th July: Tasks
- [ ] Refactor model `eos2thm`: https://github.com/ersilia-os/eos2thm/issues/6
- [ ] Refactor model `eos8a4x`: https://github.com/ersilia-os/eos8a4x/issues/6
- [ ] Test model `eos7qga`: https://github.com/ersilia-os/eos7qga/issues/4
- [ ] Re-test model `eos3cf4`: https://github.com/ersilia-os/eos3cf4/issues/3

**Comment by GemmaTuron** [comment]
Hi @HellenNamulinda !

I think we can safely merge the PR on eos2thm. I've assigned you two new models!

**Comment by HellenNamulinda** [comment]
Hi @GemmaTuron 
### Friday 7th July: Tasks 
- [ ] Debug arm64 build for model `eos8a4x`: https://github.com/ersilia-os/eos8a4x/issues/6; made a PR, https://github.com/ersilia-os/eos8a4x/pull/8
- [ ] Refactor model `eos2thm`: https://github.com/ersilia-os/eos2thm/issues/6; created a PR, https://github.com/ersilia-os/eos2thm/pull/7 
- [ ] Refactor model `eos4u6p`: https://github.com/ersilia-os/eos4u6p/issues/1
- [ ] Refactor model `eos7a04`: https://github.com/ersilia-os/eos7a04/issues/6

**Comment by HellenNamulinda** [comment]
Hi @GemmaTuron 
### Monday 10th July Tasks   
- [ ] Refactor model `eos4u6p`: https://github.com/ersilia-os/eos4u6p/issues/1; created a PR, https://github.com/ersilia-os/eos4u6p/pull/2
- [ ] Refactor model `eos7a04`: https://github.com/ersilia-os/eos7a04/issues/6
- [ ] Test Model `eos3ae6` using Colab, CLI and Docker: https://github.com/ersilia-os/eos3ae6/issues/6
- [ ] Test Model `eos1amr` using Colab, CLI and Docker: https://github.com/ersilia-os/eos1amr/issues/4

All the models tested(eos3ae6 and eos1amr)  work well using Colab and Docker. model eos1amr despite working well with string inputs on CLI, it was raising  a `TypeError: object of type 'float' has no len()` for file outputs. Model eos3ae6 works well using the three.

Done with refactoring model `eos4u6p` and created a PR. 
For model eos7a04, I'm still fixing package [version conflicts](https://github.com/ersilia-os/eos7a04/issues/6#issuecomment-1629909641).

**Comment by HellenNamulinda** [comment]
Hi @GemmaTuron 
### Tuesday 11th July Tasks  
- [ ] Refactor model `eos7a04`: https://github.com/ersilia-os/eos7a04/issues/6
- [ ] Test Model `eos6m4j` using Colab, CLI and Docker: https://github.com/ersilia-os/eos6m4j/issues/3#issuecomment-1631652077
- [ ] Test Model `eos24ci` using Colab, CLI and Docker: https://github.com/ersilia-os/eos24ci/issues/8#issuecomment-1631658191

The two models tested(eos6m4j and eos24ci)  work well using Colab, CLI and Docker.

**Comment by GemmaTuron** [comment]
Thanks for the update @HellenNamulinda 

There is still the issue in eos6m4j where some mols cannot be predicted? Also please confirm eos7w6n works now
I'll assign you new models meanwhile.

**Comment by HellenNamulinda** [comment]
Hi @GemmaTuron 
### Wenesday 12th July Tasks   
- [x] Test Model `eos7w6n` using Colab, CLI and Docker: https://github.com/ersilia-os/eos7w6n/issues/7#issuecomment-1633091343
- [x] Retest Model `eos2re5` :  https://github.com/ersilia-os/eos2re5/issues/4#issuecomment-1632869397

Model `eos7w6n`  works well using Colab, CLI and Docker.  However, for model `eos2re5`, the first value in the output(its column name: smiles) is null. Probably because it's a string yet output type is Float.

**Comment by HellenNamulinda** [comment]
Hi @GemmaTuron 
### Thursday 13th July Tasks   
- [ ] 1:1 Meeting with my mentor
- [ ]  Attend Weekly Lab Meeting
- [ ] Re-test model `eos1amr` using CLI for file outputs: https://github.com/ersilia-os/eos1amr/issues/4#issuecomment-1632477537
- [ ] Test model `eos5jz9` : https://github.com/ersilia-os/eos5jz9/issues/2
- [ ] Refactor model `eos9f6t`:  https://github.com/ersilia-os/eos9f6t/issues/1
- [ ] Refactor model `eos238c`: https://github.com/ersilia-os/eos238c/issues/1

**Comment by GemmaTuron** [comment]
@HellenNamulinda 

Great, I will not be assigning new models since you have a few open already, good work.

**Comment by HellenNamulinda** [comment]
Hello @GemmaTuron,  Sorry for the late report. I thought I had added the Friday tasks already.
### Friday 14th July Tasks    
- [ ] Tested model `eos4cxk`: https://github.com/ersilia-os/eos4cxk/issues/4
- [ ] Refactored model `eos238c`: https://github.com/ersilia-os/eos238c/issues/, Created a PR; https://github.com/ersilia-os/eos238c/pull/2

**Comment by HellenNamulinda** [comment]
### Tuesday 18th July: Tasks     
- [ ] 1:1 Meeting with my mentor
- [ ] Attend weekly team meeting
- [ ] Refactor model `eos43at`: https://github.com/ersilia-os/eos43at/issues/1
- [ ] Debug model `eos9f6t`, the inconsistent output: https://github.com/ersilia-os/eos9f6t/issues/1#issuecomment-1639645436

**Comment by HellenNamulinda** [comment]
### Thursday 20th July: Tasks      
- [ ] Finalize the presentation for and attend the weekly Thursday Lab meeting
- [ ] Refactor model `eos43at`: https://github.com/ersilia-os/eos43at/issues/1#issuecomment-1643039571, submitted a PR, https://github.com/ersilia-os/eos43at/pull/3
- [ ] Retest model `eos7pw8`: https://github.com/ersilia-os/eos7pw8/issues/9#issuecomment-1640662381
- [ ]  Refactor model `eos6fza`: https://github.com/ersilia-os/eos6fza/issues/1

**Comment by HellenNamulinda** [comment]
Hi @GemmaTuron,
### Friday 21st July: Tasks      
- [x] Setting up the [chembl_ml_tools package](https://github.com/ersilia-os/chembl_ml_tools), including installing the ChEMBL database
- [ ] Debug model `eos7pw8`, fetching from github/S3: https://github.com/ersilia-os/eos7pw8/issues/9#issuecomment-1645482014, created a PR; https://github.com/ersilia-os/eos7pw8/pull/20
- [ ]  Debug model `eos43at` : Model test on PR failing, https://github.com/ersilia-os/eos43at/pull/3#issuecomment-1645579146 and yet the model works well locally
- [ ] Refactor model `eos6fza`: https://github.com/ersilia-os/eos6fza/issues/1

I spent some time resolving `ERROR: The certificate of ‘anaconda.org’ has expired.` with model `eos7pw8` when using CLI to fetch from S3, github and repo_path(locally). The model was using `Mode: docker` when installing packaes, I configured the default  to conda. All details are explained in [the comment](https://github.com/ersilia-os/eos7pw8/issues/9#issuecomment-1645482014). Another issue with the model was `TypeError: object of type 'float' has no len()` for csv file outputs. More explanation on how it was resolved [here](https://github.com/ersilia-os/eos7pw8/issues/9#issuecomment-1646051225)

Also, to extract assays from the that will be used to test the EnsembleTabPFN package, I successfully set up the [chembl_ml_tools package](https://github.com/ersilia-os/chembl_ml_tools), including a postgres database server containing the ChEMBL database. I installed [the latest ChEMBL database] (https://ftp.ebi.ac.uk/pub/databases/chembl/ChEMBLdb/latest/).

I will now explore model `eos43at` more by testing it on Codespaces. 
Plus refactoring model `eos6fza`

**Comment by GemmaTuron** [comment]
@HellenNamulinda 

Thanks, I won't assign new tasks so you can focus on the current ones!

**Comment by HellenNamulinda** [comment]
### 26th July: Tasks       
- [ ] Refactor model `eos7jlv`: https://github.com/ersilia-os/eos7jlv/issues/1
- [ ] Refactor model `eos6fza`: https://github.com/ersilia-os/eos6fza/issues/1
- [ ] Test model `eos77w8`: https://github.com/ersilia-os/eos77w8/issues/6

**Comment by HellenNamulinda** [comment]
### 27th July: Tasks       
- [ ] Attended the weekly lab meeting
- [ ] Start work on categorizing the assays extracted from the ChEMBL database based on their standard types like; IC50, MIC, Ki, ...

**Comment by GemmaTuron** [comment]
@HellenNamulinda 

I think you have still quite some models pending to be refactored, which is probably my fault assigning too many at once. Since you are also busy with the ChEMBL data, would you tell me which models you have not started to work on, so I might re-assign them and free up some of your tasks?

**Comment by HellenNamulinda** [comment]
Hi @GemmaTuron,
It's my fault for not updating the issues for some days now. Apologies.
Let me ensure to complete all the pending by week's end.

**Comment by HellenNamulinda** [comment]
### Tuesday 1st August: Tasks     
- [x] Had a 1:1 Meeting with my mentor
- [x] Attended the weekly team meeting
- [x] Debug model `eos9f6t`: https://github.com/ersilia-os/eos9f6t/issues/1#issuecomment-1660067794; made a PR, https://github.com/ersilia-os/eos9f6t/pull/5
- [x] Refactor model `eos7jlv`: https://github.com/ersilia-os/eos7jlv/issues/1#issuecomment-1660206051; created a PR, https://github.com/ersilia-os/eos7jlv/pull/3
- [x] Tested model `eos77w8`:  https://github.com/ersilia-os/eos77w8/issues/6#issuecomment-1661382362

**Comment by HellenNamulinda** [comment]
### Wednesday 2nd August: Tasks 
- [ ] Attend the weekly lab meeting

**Model Testing:**
- [ ] Re-test model `eos2re5`: https://github.com/ersilia-os/eos2re5/issues/4#issuecomment-1659699877
- [ ] Re-test model `eos1amr`: https://github.com/ersilia-os/eos1amr/issues/4#issuecomment-1659649943
- [ ] Test model `eos1bba`: https://github.com/ersilia-os/eos1bba/issues/21
- [ ] Test model `eos4q1a`: https://github.com/ersilia-os/eos4q1a/issues/10

**Model Refactoring**
- [ ] Debug model `eos43at`: https://github.com/ersilia-os/eos43at/issues/1#issuecomment-1659714552
- [ ] Refactor model `eos935d`: https://github.com/ersilia-os/eos935d/issues/6

**Comment by HellenNamulinda** [comment]
### Thursday 3rd August: Tasks 
- [ ]  Explore and analyze the categorized ChEMBL data
- **Model Testing:**
    - [ ] Test model `eos4b8j`: https://github.com/ersilia-os/eos4b8j/issues/2

- **Model Refactoring** 
    - [ ] Refactor model `eos935d`: https://github.com/ersilia-os/eos935d/issues/6

**Comment by HellenNamulinda** [comment]
@GemmaTuron,
There is no other model.

**Comment by GemmaTuron** [comment]
Perfect, let's use today's meeting to focus on the ChEMBL data then

**Comment by HellenNamulinda** [comment]
### Tuesday 8th August: Tasks     
- [x] 1:1 Meeting with my mentor
- [x] Attend weekly team meeting
- [ ] Test Model `eos6fza`: https://github.com/ersilia-os/eos6fza/issues/3

**Comment by GemmaTuron** [comment]
Hi @HellenNamulinda 

In addition to the model testing and working on model eos96ia to help Riley, please:
- Try out the test module developed by Riley and Febie and provide feedback in the Slack thread, specifying which model did you test, and if you have any further suggestions for improvement. 
- Let me know when you want me to revise the ChEMBL data

**Comment by HellenNamulinda** [comment]
Hello @GemmaTuron,
I'm going to work on these.
Model eos96ia is one of the molgrad models and therefore will have the same docker file as eos43at. I'm testing this locally and will have a proper discussion for these molgrad models with Riley(eos96ia), Feb(eos1af5) and Zakia(eos6ao8).

Thank you!

**Comment by HellenNamulinda** [comment]
### Thursday 10th August: Tasks     
- [ ] Test/Debug model  `eos96ia`: https://github.com/ersilia-os/eos96ia/issues/3#issuecomment-1672098391
- [ ] Refactor and debug model `eos935d`: https://github.com/ersilia-os/eos935d/issues/6#issuecomment-1673228127, made a PR; https://github.com/ersilia-os/eos935d/pull/9
- [ ] Re-test Model `eos6fza`: https://github.com/ersilia-os/eos6fza/issues/5
- [ ] Present progress at the Weekly Lab meeting

**Comment by HellenNamulinda** [comment]
### Friday 11th August: Tasks     
- [x] Testing the Ersilia test module: This was done on two models; eos43at([eos43at_test.txt](https://github.com/ersilia-os/ersilia/files/12328517/eos43at_test.txt)) , and eos6f7za([eos6fza_test.txt](https://github.com/ersilia-os/ersilia/files/12328518/eos6fza_test.txt))
- [ ] ChEMBL data cleaning.

**Comment by GemmaTuron** [comment]
perfect @HellenNamulinda 

Let's try to meet on Monday to discuss the ChEMBL data!

**Comment by HellenNamulinda** [comment]
### Monday 14th & Tuesday 15th, August: Tasks 
- [x] Meeting with my mentor, Gemma, to discuss the next steps for the ChEMBL data.
- [ ] Prepare ML datasets from the ChEMBL data using the example automated pipeline, [antimicrobial-ml-tasks](https://github.com/ersilia-os/antimicrobial-ml-tasks)
- [ ] Attend weekly team meeting

**Comment by HellenNamulinda** [comment]
### Wednesday 16th, August: Tasks 
- [ ] Re-test Model `eos6fza`: https://github.com/ersilia-os/eos6fza/issues/5#issuecomment-1679650479
- [ ] Compare ChEMBL datasets generated using the [automated pipeline](https://github.com/ersilia-os/antimicrobial-ml-tasks) with the data prepared manually.

**Comment by HellenNamulinda** [comment]
### Thursday 17th August: Tasks     
- [x] Test Model `eos5qfo: https://github.com/ersilia-os/eos5qfo/issues/3#issuecomment-1681848301
- [x] Final team Internship meeting

**Comment by GemmaTuron** [comment]
Hi @HellenNamulinda !

It was great to work with you, thanks so much for your contributions to Ersilia, we hoped you learnt and enjoyed as much as we did! Please, remain engaged with the community and feel free to open any issues or contribute to open ones :)
I'll now close this issue as completed !
